### PR TITLE
Add pagination support in core for both documents and tables.

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR aims to establish a uniform pagination experience across documents and tables in core. Pagination is now optional for core endpoints, the table listing logic needed a bit refactor. This change allows view filter parameters to be applied during the query instead of post-query, which would otherwise break pagination. As a result, both the list tables and documents endpoints now return pagination cursor details, including total, limit, and offset.

## Risk

Worst case it breaks content node listing for static data sources. Safe to rollback.

### CODE REVIEW

File: core/src/stores/store.rs [hunk 1]
```diff
@@ -229,6 +230,8 @@ pub trait Store {
         &self,
         project: &Project,
         data_source_id: &str,
```
Comment (spolu) [inline core/src/stores/store.rs:232]: order of arguments should be coherent between these two functions ideally

**Review by spolu — APPROVED** [review]
LGTM

File: core/src/stores/store.rs (hunk 1 above)
Comment (flvndvd) [reply core/src/stores/store.rs:232]: Yep let me do it.

File: types/src/front/lib/core_api.ts [hunk 2]
```diff
@@ -670,35 +670,42 @@ export class CoreAPI {
...
+      documentIds,
+      projectId,
+      viewFilter,
+    }: {
+      dataSourceId: string;
+      documentIds?: string[];
+      projectId: string;
+      viewFilter?: CoreAPISearchFilter | null;
+    },
+    pagination?: { limit: number; offset: number }
```
Comment (JulesBelveze) [inline types/src/front/lib/core_api.ts:685]: `nit`: could we use `OffsetPaginationParams` here?

**Review by JulesBelveze — APPROVED** [review]

File: types/src/front/lib/core_api.ts (hunk 2 above)
Comment (flvndvd) [reply types/src/front/lib/core_api.ts:685]: It lives in front for now, but will move it in the follow up PR.

**Pull request merged**